infrastructure/dal/user_database_connector.go

```diff
@@ -73,6 +72,20 @@ func (d *DatabaseConnector) IsUserExist(data domain.UserData) (bool, error) {
 	return false, nil
 }
 
+func (d *DatabaseConnector) GetUserByLogin(login string) (domain.UserData, error) {
+	var users []UserDBDTO
+
+	err := d.Database.Select(&users, "select userlogin, userpassword from users where userlogin = $1", login)
+	if err != nil {
+		log.Println("In dal.etUserByLogin", err)
+		return domain.UserData{}, err
+	}
+	if users != nil && len(users) == 1 {
+		return UserDBDTOToUserData(users[0]), nil
+	}
+	return domain.UserData{}, domain.InvalidLogin
```

Comment:
If we get more than one record we also shall return error.
It is strange case, but we shall handle it